### ISSUE TITLE
credentials_file_path should not be required

### ISSUE THREAD

**Issue by TV4Fun** [terraform-google-modules/terraform-google-secret#4]
The `credentials` argument in the `google` provider is optional. With it unspecified, it defaults to the `gcloud` default application key. By make it required here, you are needlessly reducing the functionality of this module.

**Comment by morgante** [comment]
Unfortunately you cannot use the gcloud credentials to fetch GCS objects with Terraform: https://www.terraform.io/docs/providers/google/d/signed_url.html#credentials

**Comment by TV4Fun** [comment]
Yes, I realized that after trying it. I think there are still valid use cases for having a default value though, like if you are running on a repo managed with Atlantis.

**Comment by morgante** [comment]
I don't quite follow - what default value would you suggest? The issue is that we cannot use the default credentials file.

**Comment by TV4Fun** [comment]
The default credentials file, which might be specified in the `GOOGLE_APPLICATION_CREDENTIALS` environment or come from another data source, will work if it is for a service account, which would be the typical case if this was running in a CI configuration like Atlantis, and would save some extra manual configuration.

**Comment by morgante** [comment]
Got it, that makes sense (provided we have an explanation).
